pkg/cmd/issue/edit/edit.go

```diff
@@ -141,75 +149,107 @@ func editRun(opts *EditOptions) error {
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	issueURLs := make(chan string, len(issues))
+	if opts.workers == 0 {
+		opts.workers = 10
 	}
+	g := errgroup.Group{}
+	g.SetLimit(opts.workers)
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for i := range issues {
+		// Copy variables to capture in the go routine below.
+		issue := issues[i]
+		editable := editable.Clone()
+
+		g.Go(func() error {
+			editable.Title.Default = issue.Title
+			editable.Body.Default = issue.Body
+			editable.Assignees.Default = issue.Assignees.Logins()
+			editable.Labels.Default = issue.Labels.Names()
+			editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
+			projectItems := map[string]string{}
+			for _, n := range issue.ProjectItems.Nodes {
+				projectItems[n.Project.ID] = n.ID
+			}
+			editable.Projects.ProjectItems = projectItems
+			if issue.Milestone != nil {
+				editable.Milestone.Default = issue.Milestone.Title
+			}
+
+			// Allow interactive prompts for one issue; failed earlier if multiple issues specified.
+			if opts.Interactive {
+				editorCommand, err := opts.DetermineEditor()
+				if err != nil {
+					return err
+				}
+				err = opts.EditFieldsSurvey(&editable, editorCommand)
+				if err != nil {
+					return err
+				}
+			}
+
+			err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			if err != nil {
+				return err
+			}
 
-	opts.IO.StartProgressIndicator()
-	err = prShared.UpdateIssue(httpClient, repo, issue.ID, issue.IsPullRequest(), editable)
+			issueURLs <- issue.URL
+			return nil
+		})
+	}
 	opts.IO.StopProgressIndicator()
+
+	err = g.Wait()
+	close(issueURLs)
+
 	if err != nil {
 		return err
 	}
 
-	fmt.Fprintln(opts.IO.Out, issue.URL)
+	for issueURL := range issueURLs {
+		fmt.Fprintln(opts.IO.Out, issueURL)
```

Comment:
If we sort the issueURLs before they are output we can control the order of the output for testing and we can do away with setting the number of workers.

Reply:
I am of the mind that we want the tests to actually execute the functionality with parallelization so that we can discover any data race bugs that might be occurring that we wouldn't see by limiting to 1 worker.

Reply:
Fair. I was considering sorting originally anyway, but wasn't sure it should be a lexical sort or order based on input. I'll go lexical for now unless you prefer specified ordered.

Reply:
To note, until recently when I had to dig more into the request mocks I thought they were actually ordered too, since they tended to be registered in order (which makes sense to "visualize" at least).